src/server/router.rs

```diff
@@ -47,40 +43,38 @@ pub async fn robots_data(_req: HttpRequest) -> Result<HttpResponse, Box<dyn std:
     let page_content: String =
         read_to_string(format!("{}/robots.txt", file_path(FileType::Theme)?))?;
     Ok(HttpResponse::Ok()
-        .content_type("text/plain; charset=ascii")
+        .content_type(ContentType::plaintext())
         .body(page_content))
 }
 
 /// Handles the route of about page of the `websurfx` meta search engine website.
 #[get("/about")]
 pub async fn about(config: web::Data<Config>) -> Result<HttpResponse, Box<dyn std::error::Error>> {
-    Ok(HttpResponse::Ok()
-        .content_type("text/html; charset=utf-8")
-        .body(
-            crate::templates::views::about::about(
-                &config.style.colorscheme,
-                &config.style.theme,
-                &config.style.animation,
-            )
-            .0,
-        ))
+    Ok(HttpResponse::Ok().content_type(ContentType::html()).body(
+        crate::templates::views::about::about(
+            &config.style.colorscheme,
+            &config.style.theme,
+            &config.style.animation,
+        )
+        .0,
+    ))
 }
 
 /// Handles the route of settings page of the `websurfx` meta search engine website.
 #[get("/settings")]
 pub async fn settings(
     config: web::Data<Config>,
 ) -> Result<HttpResponse, Box<dyn std::error::Error>> {
-    Ok(HttpResponse::Ok()
-        .content_type("text/html; charset=utf-8")
-        .body(
-            crate::templates::views::settings::settings(
-                config.safe_search,
-                &config.style.colorscheme,
-                &config.style.theme,
-                &config.style.animation,
-                &config.upstream_search_engines,
-            )?
-            .0,
-        ))
+    Ok(HttpResponse::Ok().content_type(ContentType::html()).body(
+        crate::templates::views::settings::settings(
+            &config.style.colorscheme,
+            &config.style.theme,
+            &config.style.animation,
+            &config
+                .upstream_search_engines
+                .keys()
+                .collect::<Vec<&String>>(),
+        )?
+        .0,
+    ))
```

Comment:
```suggestion
    Ok(HttpResponse::Ok().content_type(ContentType::html()).body(
        crate::templates::views::settings::settings(
            config.safe_search,
            &config.style.colorscheme,
            &config.style.theme,
            &config.style.animation,
            &config.upstream_search_engines,
        )?
        .0,
    ))
```

We are sorry for this. Actually we forgot to update the issue but here is what you will need to replace the previous code with and after that it should work. So we would suggest updating the code with the suggested changes. :slightly_smiling_face:

Reply:
Thank you for explanations! Make necessary changes